src/main/java/org/jenkinsci/test/acceptance/controller/LocalController.java

```diff
@@ -374,12 +375,12 @@ private void diagnoseFailedLoad(Exception cause) throws IOException {
             }
 
             try {
-                Process jstack = new ProcessBuilder("jstack", String.valueOf(pid)).start();
-                if (jstack.waitFor() == 0) {
-                    StringWriter writer = new StringWriter();
-                    IOUtils.copy(jstack.getInputStream(), writer);
-                    return writer.toString();
-                }
+                ThreadDumpRuntime runtime = new PidRuntimeFactory().fromProcess(pid);
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                PrintStream printStream = new PrintStream(baos);
+                runtime.toString(printStream, ModelObject.Mode.MACHINE);
+                printStream.close();
```

Comment:
The only point for that stream to exist is to collect the output. `close()` effectively flushes the writer to ensure whole content is written to `toString()` returns everything. If exception gets propagated we do no longer care for the output so flushing is not necessary.